src/nb_type.cpp

```diff
@@ -662,9 +659,16 @@ PyObject *nb_type_new(const type_init_data *t) noexcept {
         }
     }
 
-    if (temp_tp->tp_members) {
+    // CPython copies member definitions into the heap type object,
+    // presumably to support something related to __slots__. PyPy doesn't,
+    // and without the Py_SIZE != 0 guard here we can wind up with
+    // tp->tp_members pointing to garbage. We only use tp_members to
+    // set __dictoffset__, and we already copied that above, so there's
+    // no harm in leaving the final tp_members set to nullptr on PyPy.
+    if (temp_tp->tp_members && Py_SIZE(temp_tp) != 0) {
         tp->tp_members = (PyMemberDef*)((char *)tp + Py_TYPE(tp)->tp_basicsize);
-        std::memcpy(tp->tp_members, temp_tp->tp_members, tp->tp_itemsize * Py_SIZE(temp_tp));
```

Comment:
On PyPy I was seeing:
* `tp->tp_itemsize == 0`
* `Py_TYPE(tp)->tp_itemsize == 0`
* `Py_SIZE(temp_tp) == 0`
so this memcpy wasn't copying anything. AFAICT it only worked in the past because `tp->tp_members` happened to point at a zero, so PyType_Ready didn't think there were any members. Something about this change tickled that uninitialized value to not be zero anymore, resulting in a segfault.